wideq/core.py

```diff
@@ -1,183 +1,70 @@
-    level = logging.INFO
-    fmt = "%(asctime)s %(levelname)s [%(name)s] %(message)s"
-    datefmt = "%Y-%m-%d %H:%M:%S"
-    logger = logging.getLogger("wideq")
-    logger.setLevel(level)
-
-    try:
-        import colorlog  # type: ignore
-        colorfmt = f"%(log_color)s{fmt}%(reset)s"
-        handler = colorlog.StreamHandler()
-        handler.setFormatter(
-            colorlog.ColoredFormatter(
-                colorfmt,
-                datefmt=datefmt,
-                reset=True,
-                log_colors={
-                    "DEBUG": "cyan",
-                    "INFO": "green",
-                    "WARNING": "yellow",
-                    "ERROR": "red",
-                    "CRITICAL": "red",
-                },
-            )
-        )
-    except ImportError:
-        handler = logging.StreamHandler()
-        handler.setFormatter(logging.Formatter(fmt=fmt, datefmt=datefmt))
-
-    logger.addHandler(handler)
-    return logger
+from datetime import datetime
+from typing import Any, Dict, Generator, Optional
 
+from . import as_list, gen_uuid
+from . import core_exceptions as exc
+from .device import DeviceInfo, DEFAULT_TIMEOUT, DEFAULT_REFRESH_TIMEOUT
 
-LOGGER = get_wideq_logger()
 
+class Tlsv1HttpAdapter(HTTPAdapter):
+    def init_poolmanager(self, connections, maxsize, block=False):
+        self.poolmanager = PoolManager(
+            num_pools=connections,
+            maxsize=maxsize,
+            block=block,
+            ssl_version=ssl.PROTOCOL_TLSv1,
+        )
 
-def retry_session():
-    """Get a Requests session that retries HTTP and HTTPS requests.
-    """
-    # Adapted from:
-    # https://www.peterbe.com/plog/best-practice-with-retries-with-requests
-    session = requests.Session()
-    retry = Retry(
-        total=RETRY_COUNT,
-        read=RETRY_COUNT,
-        connect=RETRY_COUNT,
-        backoff_factor=RETRY_FACTOR,
-        status_forcelist=RETRY_STATUSES,
-    )
-    adapter = HTTPAdapter(max_retries=retry)
-    session.mount('http://', adapter)
-    session.mount('https://', adapter)
-    return session
 
+GATEWAY_URL = "https://kic.lgthinq.com:46030/api/common/gatewayUriList"
+APP_KEY = "wideq"
+SECURITY_KEY = "nuts_securitykey"
+DATA_ROOT = "lgedmRoot"
+SVC_CODE = "SVC202"
+CLIENT_ID = "LGAO221A02"
+OAUTH_SECRET_KEY = "c053c2a6ddeb7ad97cb0eed0dcb31cf8"
+OAUTH_CLIENT_KEY = "LGAO221A02"
+DATE_FORMAT = "%a, %d %b %Y %H:%M:%S +0000"
 
-def set_log_level(level: int):
-    logger = get_wideq_logger()
-    logger.setLevel(level)
+DEFAULT_COUNTRY = "US"
+DEFAULT_LANGUAGE = "en-US"
 
+API_ERRORS = {
+    "0102": exc.NotLoggedInError,
+    "0106": exc.NotConnectedError,
+    "0100": exc.FailedRequestError,
+    9000: exc.InvalidRequestError,  # Surprisingly, an integer (not a string).
+}
 
-def gen_uuid() -> str:
-    return str(uuid.uuid4())
+_LOGGER = logging.getLogger(__name__)
 
 
-def oauth2_signature(message: str, secret: str) -> bytes:
+def oauth2_signature(message, secret):
```

Comment:
Here's an example of a change we can just revert back to the current version (since it seems to be newer).